library/ssl_srv.c

```diff
@@ -1354,8 +1354,8 @@ static int ssl_parse_client_hello_v2( mbedtls_ssl_context *ssl )
 #endif
         {
             if( p[0] != 0 ||
-                p[1] != ( ( ciphersuites[i] >> 8 ) & 0xFF ) ||
-                p[2] != ( ( ciphersuites[i]      ) & 0xFF ) )
+                p[1] != MBEDTLS_BYTE_1( ciphersuites[i] ) ||
+                p[2] != MBEDTLS_BYTE_0( ciphersuites[i] ) )
```

Comment:
Could do a similar transformation here - `MBEDTLS_GET_UINT16_BE(p, 1) != ciphersuites[i]`. Also later in 2089-2090, 2208-2209.

Reply:
Is this okay to do in the case of logical OR? `||`